### PR TITLE
Implement Wayfinder 2.1 

### LINKED ISSUE #515: Fix (or remove) scrolljacking
Scrolling on data narratives is often inconsistent and jumpy, resulting in a poor user experience even on desktop. This sometimes results in the user being unable to proceed through the page as intended. 

[screencap tbh] 

#### Solution 1: Remove completely  
- Page would no longer attempt to snap to the top of any given section, allowing for completely free/fluid scrolling. 
- Sections should still be considered distinct from one another; other features below should still work as originally intended: 
    - Each section should still have a distinct URL, which changes dynamically as the user scrolls through the page. 
    - The Wayfinder should still work as originally intended, and allow a user to jump to desired sections. 

#### Solution 2: Attempt to fix
Intended behavior would be for a single swipe or scroll to automatically move the user to the next section of the narrative, unless scrolling within the section. 
This is notably a difficult problem and has already eaten a lot of time, so we should proceed here with caution and with the knowledge of previous attempts.

### LINKED ISSUE #523: Implement Wayfinder 2.1
The Wayfinder is a feature that allows users to preview the content of a given narrative. 

The current Wayfinder has a few main issues: 
- It's very small, which presents a possible accessibility failure. 
- While it allows users to preview on hover, there is no view that allows a user to see all of the content at one go. 
- It does not exist on mobile, where users might benefit greatly from the persistent context. 

### Scope 

Wayfinder 2.1 should make it significantly easier for users to dive into and navigate the narratives, including: 
- P0: Expanding the presence of the Wayfinder to a persistent bar along the bottom of the page that highlights the "forward/backwards" buttons and the title of the current narrative. 
- P0: Implementing the Wayfinder on both desktop and mobile. 
- P0: Including a "Table of Contents" that allows the user to preview all of the sections of a narrative in one go. This should be activated by clicking/tapping a dedicated button. 
- P1: Implementing a first-run experience that introduces new users to the Wayfinder. 
- P1: Nudge users to proceed to the next section after a few seconds of landing on a narrative via a visual cue on the "next" button (pulsing, flashing, tooltip, etc).

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Added new Wayfinder as requested in [Issue](https://github.com/Recidiviz/public-dashboard/issues/523), and removed scrolljacking.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #515 
Closes #523 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: spotlight-client/src/NarrativeLayout/NarrativeNavigation/ProgressBar.tsx
Comment (terryttsai) [inline spotlight-client/src/NarrativeLayout/NarrativeNavigation/ProgressBar.tsx:65]: When do we use `scrolledPercent` instead of `percent`?

Comment (terryttsai) [reply spotlight-client/src/NarrativeLayout/NarrativeNavigation/ProgressBar.tsx:65]: Personally I like the smoothly-updating progress bar with `scrolledPercent` rather than the `percent` we're passing in that's active section / total sections.

File: spotlight-client/src/NarrativeLayout/NarrativeNavigation/Wayfinder.tsx
Comment (terryttsai) [inline spotlight-client/src/NarrativeLayout/NarrativeNavigation/Wayfinder.tsx:231]: So we want to tell our users "Introducing Wayfinder 2.1"? What is Wayfinder, should users know about this?

**Review by terryttsai — COMMENTED** [review]
Just a small question otherwise it looks great!

File: spotlight-client/src/NarrativeLayout/NarrativeNavigation/ProgressBar.tsx
Comment (nasaownsky) [reply spotlight-client/src/NarrativeLayout/NarrativeNavigation/ProgressBar.tsx:65]: cc @hobuobi

File: spotlight-client/src/NarrativeLayout/NarrativeNavigation/Wayfinder.tsx
Comment (nasaownsky) [reply spotlight-client/src/NarrativeLayout/NarrativeNavigation/Wayfinder.tsx:231]: It's more like a placeholder. Final copy is up to @hobuobi

**Review by hobuobi — CHANGES_REQUESTED** [review]
Apologies for the late review, but this is generally looks great; thank you so much for the detailed work! A few comments: 
- Would make the entire bar clickable to open and close the Table of Contents, not just the pill to the left. 
- After clicking a section in the Table of Contents, the bar should close. 
- Arrow-blinking should only happen for the first section of the page, and should start 3 seconds after landing there. 
- For now, we can get rid of the introduction modal ("Introducing Wayfinder 2.2"). 
- ~~Now that we aren't automatically adjusting the position of the page, the user can sometimes find themselves in a large whitespace between sections. We should consider reducing the amount of space between sections to minimize this effect.~~ (**EDIT**: You can ignore this one; looking at some of the live Spotlight pages, I think the visualizations take up enough space to avoid this problem.) 
- The Racial Disparities is completely broken for me on all states. This might just be due to a data issue that I'm debugging on my side though, so if it's still working for others, it's probably just because of that.

**Comment by nasaownsky** [comment]
@hobuobi a few comments about requested changes:
1. Making the entire bar clickable would result next/previous buttons working incorrectly, triggering the bar after clicking on them.
2. The Racial Disparities works perfectly in demo mode, so I assume this problem is on your side.

**Comment by terryttsai** [comment]
> Making the entire bar clickable would result next/previous buttons working incorrectly, triggering the bar after clicking on them.

What if you try adding in event.stopPropagation() when handling the next/prev button events? Then that should prevent the bar from being triggered.

**Comment by nasaownsky** [comment]
> What if you try adding in event.stopPropagation() when handling the next/prev button events? Then that should prevent the bar from being triggered.

Thanks, it worked!

**Comment by terryttsai** [comment]
@nasaownsky are you talking about test errors that you're getting locally? It looks like all tests passed in your PR.

**Comment by nasaownsky** [comment]
@terryttsai Hm, strange. I think it's good to go then.

**Review by terryttsai — APPROVED** [review]

**Comment by nasaownsky** [comment]
What's your thoughts, @hobuobi?

**Review by hobuobi — APPROVED** [review]
All of this looks good to me!! Thank you so much @nasaownsky :)

**Pull request merged**
